truss/templates/server/model_wrapper.py

```
@@ -528,6 +540,40 @@ async def poll_for_environment_updates(self) -> None:
                         exc_info=errors.filter_traceback(self._model_file_name),
                     )
 
+    async def is_ready(self) -> Optional[bool]:
+        descriptor = self.model_descriptor.is_ready
+        is_ready: Optional[bool] = None
+        if not descriptor:
+            return is_ready
+        try:
```

add loading fail fast logic here:
```
        if model.load_failed:
            INFERENCE_SERVER_FAILED_FILE.touch()
            os.kill(os.getpid(), signal.SIGKILL)
```